deep_qa/data/fields/label_field.py

```diff
@@ -0,0 +1,55 @@
+from typing import Dict, List, Union
+
+from overrides import overrides
+import numpy
+
+from . import Field
+from .. import Vocabulary
+
+class LabelField(Field):
+    """
+    A ``LabelField`` is a categorical label of some kind, where the labels are either strings of
+    text or 0-indexed integers.  If the labels need indexing, we will use a :class:`Vocabulary` to
+    convert the string labels into integers.
+
+    Parameters
+    ----------
+    label : ``Union[str, int]``
+    label_namespace : ``str``, optional (default=``labels``)
+        The namespace to use for converting label strings into integers.  If you have multiple
+        different label fields in your data, you should make sure you use different namespaces for
+        each one.
+    index_labels : ``bool``, optional (default=``True``)
+        Do the labels need indexing?  If they are categorical labels, then this should be ``True``.
+        If they are already 0-indexed integers, set this to ``False``.
+    """
+    def __init__(self, label: Union[str, int], label_namespace: str='labels', index_labels: bool=True):
+        self._label = label
+        self._label_namespace = label_namespace
+        if index_labels:
+            self._label_id = None
+        else:
+            assert isinstance(label, int), "Labels must be ints if you want to skip indexing"
+            self._label_id = label
+
+    @overrides
+    def count_vocab_items(self, counter: Dict[str, Dict[str, int]]):
+        counter[self._label_namespace][self._label] += 1
+
+    @overrides
+    def index(self, vocab: Vocabulary):
+        self._label_id = vocab.get_token_index(self._label, self._label_namespace)
+
+    @overrides
+    def get_padding_lengths(self) -> Dict[str, int]:
+        return {'num_labels': self._label_id + 1}
```

Comment:
Oops, you're right.  I was originally going to use the vocab to get the number of labels (as the previous version did), but thought I could get away without it.  You're right, you can't.  I'll fix this.